src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/InterfaceMemberImpliedModifierCheckTest.java

```diff
@@ -0,0 +1,372 @@
+            "46:5: " + getCheckMessage(MSG_KEY, "public"),
+        };
+        verify(checkConfig, getPath("InputInterfaceMemberImpliedModifierNestedOnInterface.java"),
+            expected);
+    }
+
+    @Test
+    public void testNestedOnInterfaceAllowImpliedPublicStatic() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(InterfaceMemberImpliedModifierCheck.class);
+        checkConfig.addAttribute("violateImpliedPublicNested", "false");
+        checkConfig.addAttribute("violateImpliedStaticNested", "false");
+        final String[] expected = {};
+        verify(checkConfig, getPath("InputInterfaceMemberImpliedModifierNestedOnInterface.java"),
+            expected);
+    }
+
+    @Test
+    public void testNestedOnClassIgnored() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(InterfaceMemberImpliedModifierCheck.class);
+        final String[] expected = {};
+        verify(checkConfig, getPath("InputInterfaceMemberImpliedModifierNestedOnClass.java"),
+            expected);
+    }
+
+    @Test
+    public void testNestedOnInterfaceNestedNoImpliedPublicStatic() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(InterfaceMemberImpliedModifierCheck.class);
+        final String[] expected = {
+            "7:9: " + getCheckMessage(MSG_KEY, "public"),
+            "7:9: " + getCheckMessage(MSG_KEY, "static"),
+            "10:9: " + getCheckMessage(MSG_KEY, "public"),
+            "10:9: " + getCheckMessage(MSG_KEY, "static"),
+            "15:9: " + getCheckMessage(MSG_KEY, "public"),
+            "15:9: " + getCheckMessage(MSG_KEY, "static"),
+        };
+        verify(checkConfig,
+            getPath("InputInterfaceMemberImpliedModifierNestedOnInterfaceNested.java"),
+            expected);
+    }
+
+    @Test
+    public void testNestedOnClassNestedNoImpliedPublicStatic() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(InterfaceMemberImpliedModifierCheck.class);
+        final String[] expected = {
+            "7:9: " + getCheckMessage(MSG_KEY, "public"),
+            "7:9: " + getCheckMessage(MSG_KEY, "static"),
+            "10:9: " + getCheckMessage(MSG_KEY, "public"),
+            "10:9: " + getCheckMessage(MSG_KEY, "static"),
+            "15:9: " + getCheckMessage(MSG_KEY, "public"),
+            "15:9: " + getCheckMessage(MSG_KEY, "static"),
+        };
+        verify(checkConfig, getPath("InputInterfaceMemberImpliedModifierNestedOnClassNested.java"),
+            expected);
+    }
+
+    @Test
+    public void testPrivateStaticMethod() throws Exception {
+        // private methods on interfaces are Java 9 only, but checkstyle is Java 8
+        // this test exists solely to provide coverage of private methods
+        final DetailAST modPrivate = new DetailAST();
+        modPrivate.setType(TokenTypes.LITERAL_PRIVATE);
```

Comment:
You should be able create a valid Java 9 file and place it in non-compilable folder and put a marker on it that it should be compiled by Java 9.
Example: https://github.com/checkstyle/checkstyle/blob/master/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammars/java9/InputJava9TryWithResources.java#L1